### PR TITLE
Use strerror(errno) instead of %m 

### PR DESCRIPTION
**Pull request by 0x0916**
Signed-off-by: 0x0916 <w@laoqinren.net>

### CODE REVIEW

**Comment by brauner** [comment]
Unfortunately, the `%m` specifier is a GNU extension and so I'm weary to make this the default everywhere. What you could do however, is remove all `%m` references and replace them with appropriate `strerror(errno)` calls. :)

**Comment by 0x0916** [comment]
OK, I will  remove all `%m` references.

**Comment by brauner** [comment]
@0x0916, thanks. :)

**Comment by 0x0916** [comment]
@brauner update done.

File: src/tests/lxc-test-config-jump-table [hunk 1]
```diff
@@ -0,0 +1,228 @@
+#! /bin/bash
```
Comment (brauner) [inline src/tests/lxc-test-config-jump-table]: What is this file?

File: src/tests/lxc-test-parse-config-file [hunk 2]
```diff
@@ -0,0 +1,228 @@
+#! /bin/bash
```
Comment (brauner) [inline src/tests/lxc-test-parse-config-file]: Same question: What is the purpose of this file?

File: src/lxc/cgroups/cgfsng.c [hunk 3]
```diff
@@ -1254,13 +1254,15 @@ static int cgroup_rmdir(char *dirname)
 
 	if (rmdir(dirname) < 0) {
 		if (!r)
-			WARN("%s: failed to delete %s: %m", __func__, dirname);
+			WARN("%s: failed to delete %s: %s", __func__, dirname,
```
Comment (brauner) [inline src/lxc/cgroups/cgfsng.c]: Could you remove the `__func__` macro, please? Our logging function will always spew out the function name by design.

File: src/lxc/cgroups/cgfsng.c [hunk 4]
```diff
@@ -1254,13 +1254,15 @@ static int cgroup_rmdir(char *dirname)
...
+			     strerror(errno));
 		r = -1;
 	}
 
 	if (closedir(dir) < 0) {
 		if (!r)
-			WARN("%s: failed to delete %s: %m", __func__, dirname);
+			WARN("%s: failed to delete %s: %s", __func__, dirname,
```
Comment (brauner) [inline src/lxc/cgroups/cgfsng.c]: Could you remove the `__func__` macro, please? Our logging function will always spew out the function name by design.

File: src/tests/lxc-test-config-jump-table (hunk 1 above)
Comment (0x0916) [inline src/tests/lxc-test-config-jump-table]: Sorry, these two file should be added to `.gitignore`, see https://github.com/lxc/lxc/pull/1640

File: src/lxc/cgroups/cgfsng.c (hunk 3 above)
Comment (0x0916) [inline src/lxc/cgroups/cgfsng.c]: removed.

File: src/lxc/cgroups/cgfsng.c (hunk 4 above)
Comment (0x0916) [inline src/lxc/cgroups/cgfsng.c]: removed

**Comment by brauner** [comment]
jenkins: test this please

**Comment by brauner** [comment]
@0x0916 are you interested in writing another patch?

**Pull request merged**